tests/contest/contest/src/tests/delete/delete_test.rs

```diff
@@ -0,0 +1,167 @@
+use std::thread::sleep;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::tests::lifecycle::ContainerLifecycle;
+
+/// Test deleting a non-existent container
+fn delete_non_existed_container() -> TestResult {
+    let container = ContainerLifecycle::new();
+
+    match container.delete() {
+        TestResult::Failed(_) => TestResult::Passed,
+        TestResult::Passed => TestResult::Failed(anyhow!(
+            "Expected deleting a non-existent container to fail, but it succeeded"
+        )),
+        _ => TestResult::Failed(anyhow!("Unexpected test result")),
+    }
+}
+
+/// Test deleting a container in "created" state
+fn delete_created_container_test() -> TestResult {
+    let container = ContainerLifecycle::new();
+
+    // Create the container
+    match container.create() {
+        TestResult::Passed => {}
+        _ => return TestResult::Failed(anyhow!("Failed to create container")),
+    }
+
+    // Wait for state transition
+    sleep(Duration::from_secs(1));
```

Comment:
Could this be avoided by polling for some information to avoid waiting a second?

Reply:
I agree with this. So, I try to polling for the container state using the ContainerLifecycle's `state()` method instead of using `Thread.sleep()`.

Reply:
I've updated the implementation to use polling for container status monitoring, based on the approach from https://github.com/youki-dev/youki/pull/3082/commits/5921f410d51ad5b14fff70a8321d13ed3cfcd1d5.
Could you please review these changes?